sdk/src/index.ts

```diff
@@ -125,7 +125,7 @@ export interface MixinDeviceOptions<T> {
     // if the device state came from another node worker thread.
     // This should ultimately be removed at some point in the future.
     if ((this._deviceState as any).__rpcproxy_traps_all_properties && deviceManager.createDeviceState && typeof this._deviceState.id === 'string') {
-      this._deviceState = deviceManager.createDeviceState(this._deviceState.id, this._deviceState.setState);
+      this._deviceState = deviceManager.createDeviceState(this._deviceState.id, this._deviceState.setState!);
```

Comment:
Likewise I'm manually asserting this is non undefined, but I'm not sure if the original method should be non-nullable  

https://github.com/koush/scrypted/blob/8fa0da7bd812c7badbc6c2364cfbe434e491e98a/sdk/types/src/types.input.ts#L2281